File: azure-mobile-apps-quickstarts/backend/try-mobile-apps/Controllers/TodoItemController.cs
using System;
using System.Linq;
using System.Threading.Tasks;
using System.Web.Http;
using System.Web.Http.Controllers;
using System.Web.Http.OData;
using Microsoft.Azure.Mobile.Server;
using TryMobileAppsService.DataObjects;
using TryMobileAppsService.Models;

namespace TryMobileAppsService.Controllers
{
    public class TodoItemController : TableController<TodoItem>
    {
        protected override void Initialize(HttpControllerContext controllerContext)
        {
            base.Initialize(controllerContext);
            TryMobileAppsContext context = new TryMobileAppsContext();
            DomainManager = new EntityDomainManager<TodoItem>(context, Request);
        }

        // GET tables/TodoItem
        public IQueryable<TodoItem> GetAllTodoItems()
        {
            // In your production mobile service, do not evaluate the query by calling ToList() before returning 
            // the results. This is required due to SQL Compact Edition not supporting the DateTimeOffset type. 
            return Query().ToList().AsQueryable();
        }

        // GET tables/TodoItem/48D68C86-6EA6-4C25-AA33-223FC9A27959
        public SingleResult<TodoItem> GetTodoItem(string id)
        {
            return Lookup(id);
        }

        // PATCH tables/TodoItem/48D68C86-6EA6-4C25-AA33-223FC9A27959
        public Task<TodoItem> PatchTodoItem(string id, Delta<TodoItem> patch)
        {
            patch.TrySetPropertyValue("__updatedAtDateTime", DateTime.Now);
            return UpdateAsync(id, patch);
        }

        // POST tables/TodoItem
        public async Task<IHttpActionResult> PostTodoItem(TodoItem item)
        {
            TodoItem current = await InsertAsync(item);
            return CreatedAtRoute("Tables", new { id = current.Id }, current);
        }

        // DELETE tables/TodoItem/48D68C86-6EA6-4C25-AA33-223FC9A27959
        public Task DeleteTodoItem(string id)
        {
            return DeleteAsync(id);
        }
    }
}

File: azure-mobile-apps-net-server/templates/ProjectTemplate/VisualBasic/1033/Controllers/TodoItemController.vb
Imports System.Linq
Imports System.Threading.Tasks
Imports System.Web.Http
Imports System.Web.Http.Controllers
Imports System.Web.Http.OData
Imports Microsoft.Azure.Mobile.Server

Public Class TodoItemController
    Inherits TableController(Of TodoItem)

    Protected Overrides Sub Initialize(ByVal controllerContext As HttpControllerContext)
        MyBase.Initialize(controllerContext)
        Dim context As $safecontextclassname$ = New $safecontextclassname$()
        DomainManager = New EntityDomainManager(Of TodoItem)(context, Request)
    End Sub

    ' GET tables/TodoItem
    Public Function GetAllTodoItems() As IQueryable(Of TodoItem)
        Return Query()
    End Function

    ' GET tables/TodoItem/48D68C86-6EA6-4C25-AA33-223FC9A27959
    Public Function GetTodoItem(ByVal id As String) As SingleResult(Of TodoItem)
        Return Lookup(id)
    End Function

    ' PATCH tables/TodoItem/48D68C86-6EA6-4C25-AA33-223FC9A27959
    Public Function PatchTodoItem(ByVal id As String, ByVal patch As Delta(Of TodoItem)) As Task(Of TodoItem)
        Return UpdateAsync(id, patch)
    End Function

    ' POST tables/TodoItem
    Public Async Function PostTodoItem(ByVal item As TodoItem) As Task(Of IHttpActionResult)
        Dim current As TodoItem = Await InsertAsync(item)
        Return CreatedAtRoute("Tables", New With {.id = current.Id}, current)
    End Function

    ' DELETE tables/TodoItem/48D68C86-6EA6-4C25-AA33-223FC9A27959
    Public Function DeleteTodoItem(ByVal id As String) As Task
        Return DeleteAsync(id)
    End Function

End Class